core_api/src/routes/file.py

```diff
@@ -0,0 +1,118 @@
+from uuid import UUID
+
+from fastapi import HTTPException
+from pydantic import AnyHttpUrl
+
+from core_api.src.routes.sub_app import SubApp
+from redbox.models import Chunk, File, FileStatus
+
+
+class FileSubApp(SubApp):
+    def __init__(self, router, storage_handler, log, publisher, s3, env):
+        super().__init__(router)
+
+        # TODO: move some of these to the superclass if used more frequently
+        self.storage_handler = storage_handler
+        self.log = log
+        self.publisher = publisher
+        self.s3 = s3
+        self.env = env
```

Comment:
Should any of these be in the SubApp parent class instead?

Reply:
Hmmm, lets cross that bridge as we add new sub apps